pkg/client/clients.go

```diff
@@ -69,19 +71,33 @@ func (c *Clients) GetConf() *conf.SchedulerConf {
 	return c.conf
 }
 
-func (c *Clients) WaitForSync(interval time.Duration, timeout time.Duration) error {
-	return utils.WaitForCondition(func() bool {
-		// cache is re-sync'd when all informers are sync'd
-		return c.NodeInformer.Informer().HasSynced() &&
+func (c *Clients) WaitForSync() {
+	timeNow := time.Now()
+	counter := 0
+	for {
+		if c.NodeInformer.Informer().HasSynced() &&
 			c.PodInformer.Informer().HasSynced() &&
 			c.PVCInformer.Informer().HasSynced() &&
 			c.PVInformer.Informer().HasSynced() &&
 			c.StorageInformer.Informer().HasSynced() &&
 			c.ConfigMapInformer.Informer().HasSynced() &&
 			c.NamespaceInformer.Informer().HasSynced() &&
 			c.PriorityClassInformer.Informer().HasSynced() &&
-			(c.AppInformer == nil || c.AppInformer.Informer().HasSynced())
-	}, interval, timeout)
+			(c.AppInformer == nil || c.AppInformer.Informer().HasSynced()) {
+			return
+		}
+		time.Sleep(time.Second)
+		counter++
+		if counter%10 == 0 {
+			if counter > 300 {
+				log.Logger().Warn("Still waiting for informers to sync, delay seems longer than usual...",
+					zap.Duration("timeElapsed", time.Since(timeNow).Round(time.Second)))
```

Comment:
Nit: could we rename `timeNow`? It's not changing and it feels weird to constantly referenced as "now". Like "syncStartTime" or "startTime" sound much better.

Reply:
I agree, that makes sense.